cmd/traffic/cmd/manager/mutator/service.go

```diff
@@ -186,17 +185,30 @@ func serveAndWatchTLS(ctx context.Context, s *http.Server, addr string) error {
 	if err != nil {
 		return err
 	}
-	defer tcpListener.Close()
-
-	return s.Serve(
-		&tlsListener{
-			ctx:         ctx,
-			config:      tls.Config{Certificates: []tls.Certificate{cert}},
-			certPEM:     certPEM,
-			keyPEM:      keyPEM,
-			tcpListener: tcpListener,
-		},
-	)
+
+	go func() {
+		err := s.Serve(
+			&tlsListener{
+				ctx:         ctx,
+				config:      tls.Config{Certificates: []tls.Certificate{cert}},
+				certPEM:     certPEM,
+				keyPEM:      keyPEM,
+				tcpListener: tcpListener,
+			},
+		)
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			dlog.Errorf(ctx, "failed to serve: %v", err)
+		}
+	}()
+
+	<-ctx.Done()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+	if err := s.Shutdown(ctx); err != nil {
+		dlog.Errorf(ctx, "error shutting down http server: %v", err)
+		return err
+	}
+	return nil
```

Comment:
This change seems like overkill to me, and it does not take into consideration that we already have a Hard/Soft context setup active (hard to know, but it stems from the dgroup.Group returned from the `NewService` func in manager/service.go).

It should be enough to just add:
```
	go func() {
		<-ctx.Done()
		_ = s.Shutdown(dcontext.HardContext(ctx))
	}()
```
just prior to the call to `s.Serve()`. The good thing about doing that is that any error from `s.Serve()` will then be propagated to the `Group`, and hence also gracefully terminate the other routines started by that group.

Reply:
Another problem with this change is this:
```
ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
```
because that effectively remove all context values. One side effect of doing that is that logging that the subsequent `dlog.Errorf` is using, won't be configured. We use `context.WithoutCancel(ctx)` instead of `context.Background()` for situations like this.